src/components/BadTextInput.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+  <v-text-field
+    @change="change"
+    :disabled="disabled"
+    :label="label"
+    :placeholder="placeholder"
+    :id="id"
+    :value="value"
+  >
+  </v-text-field>
+</template>
+<script>
+export default {
+  name: "BadTextInput",
+  data() {
+    return {};
+  },
+  props: {
+    change: {
+      type: Function,
+      required: true
```

Comment:
Why @change has to be set?
I think this should be optional

Reply:
it is `change` handler to trigger callback when input is changed. Otherwise we won't be able to pick up changes on an input if we don't sign up for the event.

Reply:
VueJs is reactive. You don't need to trigger this event to catch the value. You can just use the v-model.

Reply:
I tried but it doesn't seem to work as is. 
`        <bad-text-input
          id="officeID"
          label="Office ID"
          placeholder="Enter office ID"
          v-model="this.officeId"
        ></bad-text-input>`

I have managed to at least bind the value like that:
<template>
  <v-text-field
    @change="update"
    :disabled="disabled"
    :label="label"
    :placeholder="placeholder"
    :id="id"
    :value="value"
  >
  </v-text-field>
</template>

But change event never happens

Reply:
Did you try with just **officeId**, instead of this.officeId?
<bad-text-input id="officeID" label="Office ID" placeholder="Enter office ID" v-model="officeId" ></bad-text-input>

You don't need to use the change event at all

Reply:
Yeah,  it works now, updated!